fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/MetadataManager.java

```diff
@@ -385,78 +393,97 @@ public void createPartition(
+                        int totalBuckets = bucketCount * (partitionNumber + 1);
+                        if (totalBuckets > maxBucketNum) {
+                            throw new TooManyBucketsException(
+                                    String.format(
+                                            "Adding partition '%s' would result in %d total buckets for table %s, exceeding the maximum of %d buckets.",
+                                            partition.getPartitionName(),
+                                            totalBuckets,
+                                            tablePath,
+                                            maxBucketNum));
+                        }
+                    } catch (TooManyBucketsException e) {
+                        throw e;
+                    } catch (Exception e) {
+                        throw new FlussRuntimeException(
+                                String.format(
+                                        "Failed to check total bucket count for table %s",
+                                        tablePath),
+                                e);
+                    }
+
+                    try {
+                        long partitionId = zookeeperClient.getPartitionIdAndIncrement();
+                        // register partition assignments to zk first
+                        zookeeperClient.registerPartitionAssignment(
+                                partitionId, partitionAssignment);
+                        // then register the partition metadata to zk
+                        zookeeperClient.registerPartition(
+                                tablePath, tableId, partitionName, partitionId);
+                        LOG.info(
+                                "Register partition {} to zookeeper for table [{}].",
+                                partitionName,
+                                tablePath);
+                    } catch (KeeperException.NodeExistsException nodeExistsException) {
+                        if (!ignoreIfExists) {
+                            throw new PartitionAlreadyExistsException(
+                                    String.format(
+                                            "Partition '%s' already exists for table %s",
+                                            partition.getPartitionQualifiedName(), tablePath));
+                        }
+                    } catch (Exception e) {
+                        throw new FlussRuntimeException(
+                                String.format(
+                                        "Register partition to zookeeper failed to create partition %s for table [%s]",
+                                        partitionName, tablePath),
+                                e);
+                    }
+                });
+        if (!lock.isLocked() && lock.getQueueLength() == 0) {
+            createPartitionLockMap.remove(physicalTablePath, lock);
```

Comment:
If any exception happens in above `inLock(xxx` cde fragement, this lock won't be removed..